### ISSUE TITLE
Visual distortion of nav route

### ISSUE THREAD

**Issue by annasoloviova** [osmandapp/Osmand#2024]
The route gets distorted, but if you zoom closer - it's fine.
![unnamed](https://cloud.githubusercontent.com/assets/12711814/11591347/b5cd290a-9a9f-11e5-8cad-53071e3d14c9.png)
![unnamed_2](https://cloud.githubusercontent.com/assets/12711814/11591348/b5cd9534-9a9f-11e5-8d98-bc0466cc4066.png)

OSMAND version 2.1.1 and  2.2.2
Sony C5 Ultra Dual
Android 5.0

**Comment by vshcherb** [comment]
Seems to be duplicate, but don't remember which issue.

**Comment by annasoloviova** [comment]
Same issue with gpx. Tablet: Telekom Puls  - Alcatel One Touch, TCT Mobile Europe SAS

Android-version: 5.0.1, build: C1WTD00, kernel-version: 3.10.54
![pic_wrong](https://cloud.githubusercontent.com/assets/12711814/12355857/2b51d4cc-bba8-11e5-9e67-605d9b3abab4.jpg)

**Comment by JeanDom** [comment]
Still the same issue, even within todays Nightly (February 2nd, 2016).

Very annoying, EVEN AT CLOSE ZOOME LEVELS; makes navigation in urban areas as good as impossible!

It’s is a nasty bug.

I am checking every Nightly since some time and sincerely hope this will be solved with next release.

Seen on new tablet Lenovo Tab2 A7-10F; 
CPU: MediaTek® MT8127 1.3GHz Quad-Core 
GPU: Mali-450 MP4
OSMand+ (paid version) on Android 5

![2016_01_27_19 02 13](https://cloud.githubusercontent.com/assets/16562157/12756909/17790ace-c9d6-11e5-9a10-11723f290d75.png)
![2016_01_27_19 28 10](https://cloud.githubusercontent.com/assets/16562157/12756910/17981a54-c9d6-11e5-8f90-f18c77aca17f.png)

**Comment by Zahnstocher** [comment]
All affected devices use MediaTek platform with Mali GPU.

**Comment by black-square** [comment]
The same problem on **Xiaomi Redmi Note 2**:
- OS: Android 5.0.2 (MIUI Global 7.1 stable 7.1.3.0)
- Chipset: Mediatek MT6795 Helio X10
- CPU: Octa-core 2.0 GHz Cortex-A53 - Redmi Note 2
- GPU: PowerVR G6200 (**not a Mali GPU**)
- OSMand+

Also, the voice (**TTS-en**) of guidances is very distorted. I have the app an another device and the voice there sounds much and much more natural.

**Comment by BenyTheKid** [comment]
Same problem here. Lenovo Tab 2 A7-30HC, Android 5.0.1 and yup,Mediatek MT8382M.
![screenshot_2016-03-20-21-05-26](https://cloud.githubusercontent.com/assets/17969672/13907390/b0a01b90-eef5-11e5-9d73-267c470164d6.png)

**Comment by rododevr** [comment]
When dragging the map to the left, the route distortion decreases and disappears completely for the part of the route which is very near the left screen border.
To see it happen, watch this video: www.justmapzz.com/downloads/div/route_rendering_move_left.mp4
Vertical movements don't affect the distortion.
Could it have anything to do with the scale factor or a misinterpreted screen resolution?

**Comment by Harribo19** [comment]
Same here 
Xiaomi redmi note 2

**Comment by JeanDom** [comment]
Bug still present in today's official release (2.4.3

Is is worth still waiting for a solution or better to uninstall?

Jean

**Comment by JeanDom** [comment]
Bug still present in today's official release (2.4.3).

Bug is really ennoying 

Users are asking whether it is worth still waiting for a solulution.

**Comment by Harribo19** [comment]
I think, think, that it will be solved eventually , think in years
But dont expect this to be soon. So for now you must live with the bug.

But I really dont use Osmand for that. And if you zoom close enough it almost disappears
I still like Osmand very much.

Verzonden door TypeApp<http://www.typeapp.com/r>


Op 13 aug. 2016, bij 08:37, JeanDom <notifications@github.com<mailto:notifications@github.com>> schreef:

Bug still present in today's official release (2.4.3

Is is worth still waiting for a solution or better to uninstall?

Jean

**Comment by hvdwolf** [comment]
I fully agree that this should be solved but as a workaround you might try the following options. I have no idea whether it works as I don't have the issue myself (Samsung S5 mini).

In OsmAnd in plugins you can activate the "OsmAnd development" plugin which is the bottom one.
In there you can activate "Use OpenGL rendering". Does that make a difference?

On many phones you can in your Android settings, in Developer options set the option "force GPU rendering". Does that make a difference?
Some phones allow this ever per app via Settings -> Application management (or Application manager) -> "your app"

**Comment by vshcherb** [comment]
"Use OpenGL rendering" I'm afraid will have the same problem cause it still uses same code and today you should install different apk from latest build. I think next release we switch everything to opengl rendering and that issue will be gone there. 

BTW this is Android issue and not OsmAnd, that's why it works on most of video chipset. Might be upgrading OS could help

**Comment by sonora** [comment]
No action planned in OsmAnd, except general migration to OpenGL, which should make this go away.

**Comment by Harribo19** [comment]
I have now the latest version of Osmand+ but the problem still remains...  
Help?

**Comment by sonora** [comment]
Yes, apparently we did not perform the switch to OpenGL in this release, it looks like ...

**Comment by Harribo19** [comment]
Thanks for the answer. 
Thats a shame. 
Much phones with f.e. Mali gpu are now unable to do the rendering well. 
Hope it will be fixed sometime.

**Comment by sonora** [comment]
Let me copy @vshcherb here, not currently sure what our status is on things like this.
Thx - Hardy

**Comment by vshcherb** [comment]
I'm aware of that issue.

**Comment by landswellsong** [comment]
Hi,
Been suffering with it for long too. The phone is Xiaomi Redmi Note 2. Still reproducible on `master` built. OpenGL options in debugging don't seem to have any effect neither has switching native off.

That being said, I noticed the same effect with the regional outlines which led me to believe it has something to do with polygons having non-solid alpha fill (otherwise normal roads would be affected). This is further confirmed by how it affects the night mode, as you can see only the outline is affected (day mode `route` doesn't have a distinguishable outline as per default rendering setting):
![nightmode](https://user-images.githubusercontent.com/2434329/37724233-f8f380b6-2d38-11e8-8b9d-652f541e5f98.png)

I'd love to fix it definitely at some point, but my schedule doesn't permit me to at the moment. So I did a quick rendering style fix and it did have an effect. Here, same scene, left is default rendering style, right is my fix:
![before_after](https://user-images.githubusercontent.com/2434329/37724040-9613064c-2d38-11e8-97c5-c401d0c0d1c8.png)

Before it's conclusively fixed I guess some users might benefit from a temporary remedy. The file I used:
```XML
<renderingStyle name="Depends-template" depends="default" defaultColor="#f1eae4" version="1">
	<!-- In a dependent renderer, just specify what should change vs. the master/default renderer -->

	<renderingProperty attr="appMode" name="Rendering mode" description="Map optimization for respective User Profile"
		type="string" possibleValues="browse map,car,bicycle,pedestrian"/>

	<renderingConstant name="null" value="#00000000"/>

    <renderingAttribute name="route">
        <case color="#2a4bd1" strokeWidth="12:8" color_3="#ffde5b" color_2="#ff0000" strokeWidth_3="5:7">
            <apply_if nightMode="true" color="#ffdf3d" color_2="#00ff00" color_3="#41a6d9" strokeWidth="9:8" color_0="#b29c2b" strokeWidth_0="12:8"/>
        </case>
    </renderingAttribute>
</renderingStyle>
```
Basically copied over from `default` with alpha values removed. 

Note this only fixes the `route`, regional outlines are still broken, I was lazy to look up where their color is being set and making them solid would mess up the map even more than having the distorted.

Hope that helps. When I get some free time I can try debugging it myself, but not sure it will be in near future.
